src/hooks/useProducts.test.ts

```diff
@@ -0,0 +1,187 @@
+import { renderHook, waitFor } from "@testing-library/react";
+import useProducts from "./useProducts";
+import { ErrorProvider } from "../context/ErrorContext";
+import { PRODUCTS_ENDPOINT } from "../api/endpoints";
+import { HttpResponse, http } from "msw";
+import { server } from "../mocks/server";
+import { useError } from "./useError";
+import usePagination from "./usePagination";
+import { act } from "react";
+
+describe("useProducts", () => {
+  const initialParams = { page: 0, resetPage: () => {} };
+
+  describe("상품 목록 조회", () => {
+    it("상품 목록을 조회한다.", async () => {
+      const { result } = renderHook(() => useProducts(initialParams), {
+        wrapper: ErrorProvider,
+      });
+
+      await waitFor(() => {
+        expect(result.current.products).toHaveLength(20);
+      });
+    });
+
+    it("상품 목록 조회 중 로딩 상태", async () => {
+      const { result } = renderHook(() => useProducts(initialParams), {
+        wrapper: ErrorProvider,
+      });
+
+      await waitFor(() => {
+        expect(result.current.loading).toBe(true);
+      });
+    });
+
+    it("상품 목록 조회 중 에러 상태", async () => {
```

Comment:
여기는 에러 상태가 발생했을 때 어떤 결과값을 기대하는지 같이 명시해주시면 좋을거 같아요
```suggestion
    it("상품 목록 조회 중 에러가 발생할 경우 ~~가 된다.", async () => {
```